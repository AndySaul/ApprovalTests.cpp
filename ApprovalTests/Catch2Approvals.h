
#ifndef CATCHPLAYGROUND_CATCH2APPROVALS_H_H
#define CATCHPLAYGROUND_CATCH2APPROVALS_H_H

#include "namers/ApprovalTestNamer.h"

// <SingleHpp unalterable>
#ifdef APPROVALS_CATCH
#define CATCH_CONFIG_MAIN

#include "Catch.hpp"

struct Catch2ApprovalListener : Catch::TestEventListenerBase {
    using TestEventListenerBase::TestEventListenerBase;
    TestName currentTest;

    Catch2ApprovalListener(Catch::ReporterConfig const &_config) : Catch::TestEventListenerBase(_config) {}

    virtual void testCaseStarting(Catch::TestCaseInfo const &testInfo) override {

        currentTest.setFileName(testInfo.lineInfo.file);
        ApprovalTestNamer::currentTest(&currentTest);
    }

    virtual void testCaseEnded(Catch::TestCaseStats const &testCaseStats) override {
        while (!currentTest.sections.empty()) {
            currentTest.sections.pop_back();
        }
    }

    virtual void sectionStarting(Catch::SectionInfo const &sectionInfo) override {
        currentTest.sections.push_back(sectionInfo.name);
    }

    virtual void sectionEnded(Catch::SectionStats const &sectionStats) override {
        currentTest.sections.pop_back();
    }
};

CATCH_REGISTER_LISTENER(Catch2ApprovalListener)

#endif
// </SingleHpp>
#endif //CATCHPLAYGROUND_CATCH2APPROVALS_H_H
