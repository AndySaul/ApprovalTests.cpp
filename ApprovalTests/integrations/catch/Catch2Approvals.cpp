#include "ApprovalTests/integrations/catch/Catch2Approvals.h"

namespace ApprovalTests
{

}
