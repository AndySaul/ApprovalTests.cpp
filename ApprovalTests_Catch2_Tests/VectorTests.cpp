#include "../ApprovalTests/Catch2Approvals.h"

using namespace std;

TEST_CASE("YouCanVerifyVectors") {
    std::vector<std::string> v{"hello", "world"};
    Approvals::verifyAll<std::string>("UPPER CASE", v, [](std::string s, std::ostream& os){os << s << " => " << s;});
}
