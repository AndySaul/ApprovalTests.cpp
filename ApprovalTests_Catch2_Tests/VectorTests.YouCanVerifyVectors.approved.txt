UPPER CASE


hello => hello
world => world

